packages/testing/config/jest/api/jest-preset.js

```diff
@@ -0,0 +1,84 @@
+const path = require('path')
+
+const {
+  getPaths,
+  getApiSideDefaultBabelConfig,
+  getApiSideBabelPresets,
+  getApiSideBabelPlugins,
+} = require('@redwoodjs/internal')
+
+const rwjsPaths = getPaths()
+const NODE_MODULES_PATH = path.join(rwjsPaths.base, 'node_modules')
+
+// @TODO
+// @TODO Not sure this logic is super solid
+// @TODO
+if (process.env.SKIP_DB_PUSH !== '1') {
+  const process = require('process')
+  const path = require('path')
+  // Load dotenvs
+  require('dotenv-defaults/config')
+
+  const cacheDirDb = `file:${path.join(__dirname, '.redwood', 'test.db')}`
+  process.env.DATABASE_URL = process.env.TEST_DATABASE_URL || cacheDirDb
+
+  const execa = require('execa')
+  execa.sync(
+    `yarn rw`,
+    ['prisma db push', '--force-reset', '--accept-data-loss'],
+    {
+      cwd: rwjsPaths.api.base,
+      stdio: 'inherit',
+      shell: true,
+      env: {
+        DATABASE_URL: process.env.DATABASE_URL,
+      },
+    }
+  )
+
+  // If its been reset once, we don't need to re-run it for every test
+  process.env.SKIP_DB_PUSH = '1'
+}
+
+module.exports = {
+  // To make sure other config option which depends on rootDir use
+  // correct path, for example, coverageDirectory
+  rootDir: rwjsPaths.base,
+  roots: [path.join(rwjsPaths.api.src)],
+  testEnvironment: path.join(__dirname, './RedwoodApiJestEnv.js'),
+  displayName: {
+    color: 'redBright',
+    name: 'api',
+  },
+  collectCoverageFrom: [
+    '**/*.{js,jsx,ts,tsx}',
+    '!**/node_modules/**',
+    '!**/dist/**',
+  ],
+  coverageDirectory: path.join(rwjsPaths.base, 'coverage'),
+  watchPlugins: [
+    'jest-watch-typeahead/filename',
+    'jest-watch-typeahead/testname',
+  ],
```

Comment:
Do you think we should still keep the common config options like this and `collectCoverageFrom` in a different file and maybe just import and spread in both api web preset? For example, if we have to add a plugin, we just have to update that common file.

Reply:
I was intentionally trying to avoid splitting out this preset file - keeping it as simple as possible to avoid jest doing something weird (like we've seen with Babel setting serialisation)

I don't think a little bit of duplication is a bad thing here, but totally open to try.

Reply:
Ok with duplication as well as long as it's consistent (see my comment above about `coverageDirectory`).

Reply:
Ok with duplication as well as long as it's consistent (see my comment above about `coverageDirectory`).

Reply:
@callingmedic911 just confirming you don't feel too strongly about this? Resolving this comment

Reply:
Yeah, good with me. We can always migrate if we realise any maintenance overhead. 🚀